### PR TITLE
utils: mappings keywords extraction

### PR DESCRIPTION
**Pull request by Panos512**
- Keywords generator now also pulls keywords from the elasticsearch
  mapping file.
- Should be merged together with inspirehep/inspire-next#783 .

Signed-off-by: Panos Paparrigopoulos panos.paparrigopoulos@cern.ch

### CODE REVIEW

**Comment by Panos512** [comment]
@jmartinm This should fix your problem with the new fields

File: invenio_query_parser/utils.py [hunk 1]
```diff
@@ -68,17 +68,33 @@ def generate_keywords():
         from invenio_base.globals import cfg
```
Comment (kaplun) [inline invenio_query_parser/utils.py:67]: Can you rename the function to: something more meaningful such as `generate_valid_keywords`, and add a docstring? (I should have commented on this in previous commit, sorry! :smile: )

File: invenio_query_parser/utils.py [hunk 2]
```diff
@@ -68,17 +68,33 @@ def generate_keywords():
...
         keyword_mapping = cfg['SEARCH_ELASTIC_KEYWORD_MAPPING']
         json_schema_paths = cfg['JSON_SCHEMA_PATHS']
+        elastic_config_paths = cfg['ELASTIC_MAPPINGS_PATHS']
```
Comment (kaplun) [inline invenio_query_parser/utils.py:70]: If we reach this point of code then we are sure we are in the valid context. In order to ensure performance, at this point you can use a global variable to store the result of the whole process and return it, if it was already computed.

Comment (kaplun) [inline invenio_query_parser/utils.py:70]: These 3 variables should be defined with empty defaults in the `config.py` file of invenio-query-parser.

File: invenio_query_parser/utils.py [hunk 3]
```diff
@@ -68,17 +68,33 @@ def generate_keywords():
...
+        with open(path) as data_file:
+            data = json.load(data_file)
+        data = data.get('mappings').get('record').get('properties')
+        dotted_keywords = get_dotted_keys(data, '', [])
+        keywords += dotted_keywords
+    # Remove duplicates.
+    keyword_list = []
+    for i in keywords:
+        if i not in keyword_list:
+            keyword_list.append(i)
```
Comment (kaplun) [inline invenio_query_parser/utils.py]: You can just define `keyword` as a `set` and in this way you will not have duplicates :smile:

**Comment by Panos512** [comment]
@kaplun done! thanks for the tips

**Pull request merged**